lib/openqcd_interface.cpp

```diff
@@ -0,0 +1,1360 @@
+  {"QUDA_INVALID_ACCELERATOR",              std::to_string(QUDA_INVALID_ACCELERATOR)},
+  {"QUDA_L2_RELATIVE_RESIDUAL",             std::to_string(QUDA_L2_RELATIVE_RESIDUAL)},
+  {"QUDA_L2_ABSOLUTE_RESIDUAL",             std::to_string(QUDA_L2_ABSOLUTE_RESIDUAL)},
+  {"QUDA_HEAVY_QUARK_RESIDUAL",             std::to_string(QUDA_HEAVY_QUARK_RESIDUAL)},
+  {"QUDA_INVALID_RESIDUAL",                 std::to_string(QUDA_INVALID_RESIDUAL)},
+  {"QUDA_NULL_VECTOR_SETUP",                std::to_string(QUDA_NULL_VECTOR_SETUP)},
+  {"QUDA_TEST_VECTOR_SETUP",                std::to_string(QUDA_TEST_VECTOR_SETUP)},
+  {"QUDA_INVALID_SETUP_TYPE",               std::to_string(QUDA_INVALID_SETUP_TYPE)},
+  {"QUDA_TRANSFER_AGGREGATE",               std::to_string(QUDA_TRANSFER_AGGREGATE)},
+  {"QUDA_TRANSFER_COARSE_KD",               std::to_string(QUDA_TRANSFER_COARSE_KD)},
+  {"QUDA_TRANSFER_OPTIMIZED_KD",            std::to_string(QUDA_TRANSFER_OPTIMIZED_KD)},
+  {"QUDA_TRANSFER_OPTIMIZED_KD_DROP_LONG",  std::to_string(QUDA_TRANSFER_OPTIMIZED_KD_DROP_LONG)},
+  {"QUDA_TRANSFER_INVALID",                 std::to_string(QUDA_TRANSFER_INVALID)},
+  {"QUDA_EIG_TR_LANCZOS",                   std::to_string(QUDA_EIG_TR_LANCZOS)},
+  {"QUDA_EIG_BLK_TR_LANCZOS",               std::to_string(QUDA_EIG_BLK_TR_LANCZOS)},
+  {"QUDA_EIG_IR_ARNOLDI",                   std::to_string(QUDA_EIG_IR_ARNOLDI)},
+  {"QUDA_EIG_BLK_IR_ARNOLDI",               std::to_string(QUDA_EIG_BLK_IR_ARNOLDI)},
+  {"QUDA_EIG_INVALID",                      std::to_string(QUDA_EIG_INVALID)},
+  {"QUDA_SPECTRUM_LM_EIG",                  std::to_string(QUDA_SPECTRUM_LM_EIG)},
+  {"QUDA_SPECTRUM_SM_EIG",                  std::to_string(QUDA_SPECTRUM_SM_EIG)},
+  {"QUDA_SPECTRUM_LR_EIG",                  std::to_string(QUDA_SPECTRUM_LR_EIG)},
+  {"QUDA_SPECTRUM_SR_EIG",                  std::to_string(QUDA_SPECTRUM_SR_EIG)},
+  {"QUDA_SPECTRUM_LI_EIG",                  std::to_string(QUDA_SPECTRUM_LI_EIG)},
+  {"QUDA_SPECTRUM_SI_EIG",                  std::to_string(QUDA_SPECTRUM_SI_EIG)},
+  {"QUDA_SPECTRUM_INVALID",                 std::to_string(QUDA_SPECTRUM_INVALID)},
+  {"QUDA_MEMORY_DEVICE",                    std::to_string(QUDA_MEMORY_DEVICE)},
+  {"QUDA_MEMORY_DEVICE_PINNED",             std::to_string(QUDA_MEMORY_DEVICE_PINNED)},
+  {"QUDA_MEMORY_HOST",                      std::to_string(QUDA_MEMORY_HOST)},
+  {"QUDA_MEMORY_HOST_PINNED",               std::to_string(QUDA_MEMORY_HOST_PINNED)},
+  {"QUDA_MEMORY_MAPPED",                    std::to_string(QUDA_MEMORY_MAPPED)},
+  {"QUDA_MEMORY_MANAGED",                   std::to_string(QUDA_MEMORY_MANAGED)},
+  {"QUDA_MEMORY_INVALID",                   std::to_string(QUDA_MEMORY_INVALID)},
+  {"QUDA_CUSOLVE_EXTLIB",                   std::to_string(QUDA_CUSOLVE_EXTLIB)},
+  {"QUDA_EIGEN_EXTLIB",                     std::to_string(QUDA_EIGEN_EXTLIB)},
+  {"QUDA_EXTLIB_INVALID",                   std::to_string(QUDA_EXTLIB_INVALID)}
+};
```

Comment:
This is something we should probably standardize this for all enums and make available library wide.  @mathiaswagner did I imagine us talking about something like this a few years ago?

Reply:
Yes, we had discussed that but seems we lost track.